lib/fluent/plugin/out_datadog.rb

```diff
@@ -112,6 +119,12 @@ def write(chunk)
       if @dd_tags
         record["ddtags"] = @dd_tags
       end
+      if @service
+        record["service"] = @service
+      end
+      if @dd_hostname
```

Comment:
Won't `@dd_hostname` always be set — i.e. this condition is always true. This has undesirable side-effects in, for example, aggregation configurations, where log entries are generated in one server's `td-agent` and forwarded through another `td-agent` (an aggregator).